pkg/util/node/node_test.go

```diff
@@ -201,3 +205,266 @@ func Test_GetZoneKey(t *testing.T) {
+			expectedErr:        nil,
+		},
+		{
+			name: "patch with node addresses using replace strategy instead of merge",
+			oldNode: &v1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-node",
+				},
+				Spec: v1.NodeSpec{
+					PodCIDR: "172.17.0.1/24",
+				},
+				Status: v1.NodeStatus{
+					Conditions: []v1.NodeCondition{
+						{
+							Type:   v1.NodeReady,
+							Status: v1.ConditionTrue,
+							Reason: "KubeletReady",
+						},
+					},
+					Addresses: []v1.NodeAddress{
+						{
+							Type:    v1.NodeInternalIP,
+							Address: "10.0.0.1",
+						},
+						{
+							Type:    v1.NodeExternalIP,
+							Address: "1.1.1.1",
+						},
+					},
+				},
+			},
+			newNode: &v1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-node",
+				},
+				Spec: v1.NodeSpec{
+					PodCIDR: "172.17.0.2/24",
+				},
+				Status: v1.NodeStatus{
+					Conditions: []v1.NodeCondition{
+						{
+							Type:   v1.NodeReady,
+							Status: v1.ConditionTrue,
+							Reason: "KubeletReady",
+						},
+					},
+					Addresses: []v1.NodeAddress{
+						{
+							Type:    v1.NodeInternalIP,
+							Address: "10.0.0.1",
+						},
+						{
+							Type:    v1.NodeInternalIP,
+							Address: "10.0.0.2",
+						},
+					},
+				},
+			},
+			expectedNode: &v1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-node",
+				},
+				Spec: v1.NodeSpec{
+					PodCIDR: "172.17.0.1/24",
+				},
+				Status: v1.NodeStatus{
+					Conditions: []v1.NodeCondition{
+						{
+							Type:   v1.NodeReady,
+							Status: v1.ConditionTrue,
+							Reason: "KubeletReady",
+						},
+					},
+					Addresses: []v1.NodeAddress{
+						{
+							Type:    v1.NodeInternalIP,
+							Address: "10.0.0.1",
+						},
+						{
+							Type:    v1.NodeInternalIP,
+							Address: "10.0.0.2",
+						},
+					},
+				},
+			},
+			expectedPatchBytes: []byte(`{"status":{"addresses":[{"address":"10.0.0.1","type":"InternalIP"},{"address":"10.0.0.2","type":"InternalIP"},{"$patch":"replace"}]}}`),
+			expectedErr:        nil,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			clientset := fake.NewSimpleClientset()
+			_, err := clientset.CoreV1().Nodes().Create(context.TODO(), test.oldNode, metav1.CreateOptions{})
+			if err != nil {
+				t.Fatalf("failed to create initial node: %v", err)
+			}
+			node, patchBytes, err := PatchNodeStatus(clientset.CoreV1(), types.NodeName(test.oldNode.Name), test.oldNode, test.newNode)
+			if !reflect.DeepEqual(node, test.expectedNode) {
+				t.Logf("actual patched node: %v", node)
+				t.Logf("expected patched node: %v", test.expectedNode)
+				t.Error("unexpectd node")
+			}
+
+			if !reflect.DeepEqual(patchBytes, test.expectedPatchBytes) {
```

Comment:
Testing the returned patch bytes seems both overly fragile and unnecessary; if you got `expectedNode` back then it doesn't matter how you got there...